pkg/registry/generic/registry/store.go

```diff
@@ -45,94 +45,112 @@ import (
+	// KeyFund is dalled for Create/Update/Get/Delete. Note that 'namespace'
+	// can be gotten from ctx.
+	//
+	// KeyFunc and KeyRootFunc must be supplied together or not at all.
 	KeyFunc func(ctx genericapirequest.Context, name string) (string, error)
 
-	// Called to get the name of an object
+	// ObjectNameFunc returns the name of an object or an error.
 	ObjectNameFunc func(obj runtime.Object) (string, error)
 
-	// Return the TTL objects should be persisted with. Update is true if this
-	// is an operation against an existing object. Existing is the current TTL
-	// or the default for this operation.
+	// TTLFunc returns the TTL objects should be persisted with. Update is
+	// true if this is an operation against an existing object. Existing is
+	// the current TTL or the default for this operation.
 	TTLFunc func(obj runtime.Object, existing uint64, update bool) (uint64, error)
 
-	// Returns a matcher corresponding to the provided labels and fields.
+	// PredicateFunc returns a matcher corresponding to the provided labels
+	// and fields.  The SelectionPredicate returned should return true if the
+	// object matches the given field and label selectors.
 	PredicateFunc func(label labels.Selector, field fields.Selector) storage.SelectionPredicate
 
-	// Called to cleanup storage clients.
-	DestroyFunc func()
-
-	// EnableGarbageCollection affects the handling of Update and Delete requests. It
-	// must be synced with the corresponding flag in kube-controller-manager.
+	// EnableGarbageCollection affects the handling of Update and Delete
+	// requests. It must be synced with the corresponding flag in kube-
+	// controller-manager.
 	EnableGarbageCollection bool
 
 	// DeleteCollectionWorkers is the maximum number of workers in a single
 	// DeleteCollection call.
 	DeleteCollectionWorkers int
 
-	// Decorator is called as exit hook on object returned from the underlying storage.
-	// The returned object could be individual object (e.g. Pod) or the list type (e.g. PodList).
-	// Decorator is intended for integrations that are above storage and
-	// should only be used for specific cases where storage of the value is
-	// not appropriate, since they cannot be watched.
+	// Decorator is called as exit hook on object returned from the underlying
+	// storage. The returned object could be an individual object (e.g. Pod)
+	// or a list type (e.g. PodList). Decorator is intended for integrations
+	// that are above storage and should only be used for specific cases where
+	// storage of the value is not appropriate, since they cannot be watched.
 	Decorator rest.ObjectFunc
-	// Allows extended behavior during creation, required
+	// CreateStrategy implements resource-specific behavior during creation,
+	// required.
 	CreateStrategy rest.RESTCreateStrategy
-	// On create of an object, attempt to run a further operation.
+	// AfterCreate implements a further operation to run after a resource is
```

Comment:
Do the after funcs run after every step of the associated strategy has been run? 

i.e. for create, AfterFunc runs after prepareforcreate, validate, canonicalize?